analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/mvn/api/impl/StitchingApiServiceImpl.java

```diff
@@ -266,4 +276,17 @@ private Revision revisionIdToRevision(Collection<Revision> revisions, long id) {
         visited.remove(source);
         return vulnerablePaths;
     }
+
+    public ResponseEntity<String> batchIngestArtifacts(JSONArray jsonArtifacts) {
```

Comment:
Maybe not in this PR,  but a potential optimization is to not only have a batch endpoint but also do the insert in a batch. I.e. do the database call in batches (because those are relatively expensive). In the current way we still process every artifact individually and only the overhead of multiple REST calls is optimized.

Reply:
Fair point. We should definitely do that later.